analyzers/tests/SonarAnalyzer.Test/TestCases/PrivateFieldUsedAsLocalVariable.cs

```diff
@@ -412,6 +461,25 @@ public void Program()
 
         private void Increment(int dummy)
             => _wasCalled = true;
+    }
+
+    public class Repo_8239_Variation_Noncompliant
+    {
+        private bool _wasCalled; // Noncompliant
 
+        public void Program()
+        {
+            var list = new List<int>();
+            list.ForEach(x => Increment(x));
+            _wasCalled = false;
+
+            if (_wasCalled)
+            {
+                Console.WriteLine("OK");
+            }
+        }
+
+        private void Increment(int dummy)
```

Comment:
Is it possible ever to have such method in another class/assembly?
I know that fields should be accessible in the class it self only - but just to be sure/.

Reply:
I don't think it is possible.
If the field is updated outside the class, it is out of scope of this rule, IMO. And very peculiar, I am thinking of using `unsafe` code or reflection.

Reply:
thanks!